src/pudl/transform/eia861.py

```diff
@@ -428,6 +442,20 @@
 ###############################################################################
 # EIA Form 861 Transform Helper functions
 ###############################################################################
+def _pre_process(df: pd.DataFrame) -> pd.DataFrame:
+    """Pre-processing applied to all EIA-861 dataframes.
+
+    * Standardize common NA values found in EIA spreadsheets.
+    * Convert report_year column to report_date.
+    """
+    return fix_eia_na(df).pipe(convert_to_date)
+
+
+def _post_process(df: pd.DataFrame) -> pd.DataFrame:
+    """Post-processing applied to all EIA-861 dataframes."""
+    return convert_cols_dtypes(df, data_source="eia")
+
+
```

Comment:
The asset factories are cool, but they ended up being really inefficient since the work involved in these steps is so small. Rather than dynamically constructing pre/post processing steps in the DAG I just call these helpers in each of the transform functions.

Reply:
I wish we could select groups of assets to be executed in the same process. This would allow us to express more dependencies between assets and operations.